ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/resources/arjuna/ServerTopLevelAction.java

```diff
@@ -591,6 +591,9 @@ protected boolean registerResource (Coordinator theCoordinator)
 
 	if (theCoordinator != null)
 	{
+	    if (resourceRefRegistered)
```

Comment:
I did but I don't think I need to change it in StrictServerTLA. I was going to rerun to see the paths into ServerActions. Until I do that I believe ServerStrictTopLevelAction is for remote calls that use slot id zero whereas ServerTopLevelAction is for calls coming from WildFly and they will be different instances - ie they both need separate protection: StrictServerTLA already uses a field called _registered to decide if the resource has been registered. ServerTLA did not have similar protection and I introduced the field resourceRefRegistered to handle that.